massa-execution-worker/src/context.rs

```diff
@@ -685,6 +685,69 @@ impl ExecutionContext {
         )
     }
 
+    /// Try to slash `roll_count` rolls from the denounced address. If not enough rolls,
+    /// slash the available amount and return the result
+    ///
+    /// # Arguments
+    /// * `denounced_addr`: address to sell the rolls from
+    /// * `roll_count`: number of rolls to slash
+    pub fn try_slash_rolls(
+        &mut self,
+        denounced_addr: &Address,
+        roll_count: u64,
+    ) -> Result<Amount, ExecutionError> {
+        // try to slash as many roll as available
+        let slashed_rolls = self
+            .speculative_roll_state
+            .try_slash_rolls(denounced_addr, roll_count);
+
+        // convert slashed rolls to coins (as deferred credits => coins)
+        let mut slashed_coins = self
+            .config
+            .roll_price
+            .checked_mul_u64(slashed_rolls.unwrap_or_default())
+            .ok_or_else(|| {
+                ExecutionError::RuntimeError(format!(
+                    "Cannot multiply roll price by {}",
+                    roll_count
+                ))
+            })?;
+
+        // what remains to slash (then will try to slash as many deferred credits as avail/what remains to be slashed)
+        let amount_remaining_to_slash = self
+            .config
+            .roll_price
+            .checked_mul_u64(roll_count)
+            .ok_or_else(|| {
+                ExecutionError::RuntimeError(format!(
+                    "Cannot multiply roll price by {}",
+                    roll_count
+                ))
+            })?
+            .saturating_sub(slashed_coins);
+
+        if amount_remaining_to_slash > Amount::zero() {
+            // There is still an amount to slash for this denunciation so we need to slash
+            // in deferred credits
+            let slashed_coins_in_deferred_credits = self
+                .speculative_roll_state
+                .try_slash_deferred_credits(&self.slot, denounced_addr, &amount_remaining_to_slash);
+
+            slashed_coins = slashed_coins.saturating_add(slashed_coins_in_deferred_credits);
+            let amount_remaining_to_slash_2 =
+                slashed_coins.saturating_sub(slashed_coins_in_deferred_credits);
+            if amount_remaining_to_slash_2 > Amount::zero() {
+                // Use saturating_mul_u64 to avoid an error (for a warn!)
+                warn!("Slashed {} coins (by selling rolls) and {} coins from deferred credits but cumulative amount is lower than expected: {} coins", 
```

Comment:
add the address being slashed to the warning

Reply:
Done